include/riak_kv_sweeper.hrl

```diff
@@ -0,0 +1,65 @@
+%% Copyright (c) 2011-2015 Basho Technologies, Inc.  All Rights Reserved.
+%%
+%% This file is provided to you under the Apache License,
+%% Version 2.0 (the "License"); you may not use this file
+%% except in compliance with the License.  You may obtain
+%% a copy of the License at
+%%
+%%   http://www.apache.org/licenses/LICENSE-2.0
+%%
+%% Unless required by applicable law or agreed to in writing,
+%% software distributed under the License is distributed on an
+%% "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+%% KIND, either express or implied.  See the License for the
+%% specific language governing permissions and limitations
+%% under the License.
+
+%% Used as configuration for sweep participants.
+-record(sweep_participant,
```

Comment:
Revisiting its usage today... it should not be in its own module. It belongs in `riak_kv_sweeper_fold`. 

Outside `riak_kv_sweeper_fold` the usage is very limited and can be fixed with a few setters/getters - tests have not been considered.
```
src/riak_kv_delete_sup.erl
70:      #sweep_participant{ description = "Object TTL",
89:      #sweep_participant{ description = "Reap tombstones",

src/riak_kv_entropy_manager.erl
244:      #sweep_participant{

src/riak_kv_sweeper.erl
25:%% schedules sweeps.  Anyone can register a #sweep_participant{} with
34:%% #sweep_participant{}.
123:-spec add_sweep_participant(#sweep_participant{}) -> ok.
139:-spec status() -> {[#sweep_participant{}], [#sweep{}]}.
161:                           [#sweep_participant{}],

src/riak_kv_sweeper_cli.erl
136:     || {Index, #sweep_participant{module = Mod}} <- IndexedParticipants].
162:format_sweep_participant(Index, #sweep_participant{module = Mod,

src/riak_kv_sweeper_state.erl
100:                            Participant:: #sweep_participant{}) -> state().
135:                          {Participants :: [#sweep_participant{}],
153:    {'ok', Participants :: [#sweep_participant{}], state()}.
162:        [Participant#sweep_participant{sweep_fun = Fun, acc = InitialAcc} ||
189:                           [#sweep_participant{}],
344:     {Module, #sweep_participant{run_interval = RunInterval}}
525:        {ok, #sweep_participant{run_interval = RunInterval}} ->
676:    #sweep_participant{module = Module,
744:    #sweep_participant{
```

`riak_kv_delete_sup`: a `riak_kv_sweeper_fold:new_participant/5` function to create the record.

`riak_kv_entropy_manager`: use the function above.
 
`riak_kv_sweeper`: use the `riak_kv_sweeper_fold:participant()` type instead of the record in the type declarations.

`riak_kv_sweeper_cli`:  move the formatting to `riak_kv_sweeper_fold`.

`riak_kv_sweeper_state`: 
* Add a function to get the module out. Not sure what the name should be, maybe `participant_module/1`?
* Add a function to set the fun and initial accumulator. Say, `set_sweep_parameters/3`.
* Add a function to set the run interval.

This is doable. It will make the testing in `riak_kv_sweeper_state` a bit harder, but maybe the generators for the `#sweep_participant{}` should be defined alongside the record definition?